src/Spectre.Service.Tests/Loaders/DatasetLoaderTest.cs

```
@@ -0,0 +1,109 @@
+
+            _mockFileSystem.AddFile(Path.Combine(localDirFull, "local_correct.txt"), new MockFileData(correctDataset));
+            _mockFileSystem.AddFile(Path.Combine(remoteDirFull, "remote_correct.txt"), new MockFileData(correctDataset));
+            _mockFileSystem.AddFile(Path.Combine(localDirFull, "local_incorrect.txt"), new MockFileData(textContents: "incorrect_data"));
+            _mockFileSystem.AddFile(Path.Combine(remoteDirFull, "remote_incorrect.txt"), new MockFileData(textContents: "incorrect_data"));
+
+            _rootConfig = new DataRootConfig(localDirFull, remoteDirFull);
+            _datasetLoader = new DatasetLoader(_rootConfig);
+        }
+
+        [Test]
+        public void ReturnsFromCorrectNameLocal()
+        {
+            Assert.IsNotNull(anObject: _datasetLoader.GetFromName(name: "local_correct.txt"),
+                message: "Loader did not manage to load local file.");
+        }
+
+        [Test]
+        public void ReturnsFromCorrectNameRemote()
+        {
+            Assert.IsNotNull(anObject: _datasetLoader.GetFromName(name: "remote_correct.txt"),
+                message: "Loader did not manage to load remote file.");
+        }
+
+        [Test]
+        public void ThrowsOnIncorrectName()
+        {
+            Assert.Throws<DatasetNotFoundException>(code: () => _datasetLoader.GetFromName(name: "invalid_name"),
+                message: "Loader accepted invalid file name.");
+        }
+
+        [Test]
+        public void ThrowsOnIncorrectFileContents()
+        {
+            Assert.Throws<DatasetFormatException>(code: () => _datasetLoader.GetFromName(name: "local_incorrect.txt"),
+                message: "Loader did not manage to load remote file.");
+        }
+
+        [Test]
+        public void DeletesIncorrectFilesFromLocal()
+        {
+            try
+            {
+                _datasetLoader.GetFromName(name: "remote_incorrect.txt");
+            }
+            catch (Exception)
```

If it is ignored, we should listen for specific type of exception to not to hide errors in tests.